andela/learning-map-front
# #150001988 Create sidebar, toggle and display contribution bookmarks

#### What does this PR do?
Allows user to view sidebar and toggle the bookmarking of contributions and displays bookmark on the side bar.

#### Description of Task to be completed? 
- Create the sidebar component
- Create App Wide Menus on the sidebar
- Create bookmark section that display all user's bookmarks
- Create bookmark component that toggles the bookmarking of a contribution
- Restructure routing such that the Navbar and Sidebar appears on all pages
- Ensure Navbar and Sidebar are sticky such that scrolling of page content does not move them

#### How should this be manually tested?
After cloning the repo

- Run yarn install
- Run yarn start
- Access the landing page via localhost:3000 and click on the LOGIN TO CONTRIBUTE button.
- On the dashboard click on the hamburger icon to display the sidebar and its contents.
- Click the bookmark icon on each contribution card to toggle bookmark and see it appear and disappear on the sidebar.
- Scroll the page to see the Navbar and Sidebar remain in position
- Click the add contribution button to move another page to see the Navbar and Sidebar there.

#### Any background context you want to provide?
- There was no sidebar
- Contributions were not bookmarkable
- Navbar has to be included on each page

#### What are the relevant pivotal tracker stories?
[#150001966](https://www.pivotaltracker.com/story/show/150001966)
[#150001988](https://www.pivotaltracker.com/story/show/150001988)
[#150473913](https://www.pivotaltracker.com/story/show/150473913)

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/25608624/29749067-f5a2fa0a-8b1b-11e7-885b-4ecca409f256.png)
![image](https://user-images.githubusercontent.com/25608624/29749069-0693eaf4-8b1c-11e7-829d-d8e24fa94343.png)
![image](https://user-images.githubusercontent.com/25608624/29749070-0da58956-8b1c-11e7-9e6f-92d76cf59573.png)
![image](https://user-images.githubusercontent.com/25608624/29749072-19352a4c-8b1c-11e7-8b99-15c2b98f6285.png)
<img width="1277" alt="screen shot 2017-08-27 at 10 52 33 am" src="https://user-images.githubusercontent.com/25608624/29749088-5a1298f6-8b1c-11e7-8f9c-3539a3b89861.png">
<img width="1280" alt="screen shot 2017-08-27 at 10 53 11 am" src="https://user-images.githubusercontent.com/25608624/29749090-73870178-8b1c-11e7-8deb-d8f200a71daf.png">
<img width="1280" alt="screen shot 2017-08-27 at 10 53 41 am" src="https://user-images.githubusercontent.com/25608624/29749091-7c0d0176-8b1c-11e7-8a37-f3f88a4fd86a.png">
<img width="1273" alt="screen shot 2017-08-27 at 11 26 51 am" src="https://user-images.githubusercontent.com/25608624/29749094-81d729ba-8b1c-11e7-9e71-e6a2240e03b2.png">
